### PR TITLE
NAS-113823 / 12.0 / Make plugin jail a base jail if it's not already a basejail

### PR DESCRIPTION
**Pull request by sonicaj**
This PR adds changes for plugin jails to convert them to basejails when the plugin is upgraded if it's not already a basejail. This case wouldn't ideally happen unless for some users who are running our migrated version of asigra plugin where the jail is not a basejail.

### CODE REVIEW

File: iocage_lib/create_utils.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    for base_dir in basedirs:
+        source = os.path.join(
+            iocroot, 'templates' if all(k not in release for k in ('-RELEASE', '-STABLE')) else 'releases',
+            release, 'root', base_dir
+        )
+        destination = os.path.join(iocroot, 'jails', jail_uuid, 'root', base_dir)
+
+        # This reduces the REFER of the basejail.
+        shutil.rmtree(destination, ignore_errors=True)
```
Comment (william-gr) [inline iocage_lib/create_utils.py:25]: Why the change from rm proc to shutil?

Comment (sonicaj) [reply iocage_lib/create_utils.py:25]: Seemed cleaner to remove an extra subprocess call ? In practice didn't seem to introduce a performance regression ?
We can have it if you think it would still be nice/better :)

Comment (william-gr) [reply iocage_lib/create_utils.py:25]: One less chance of regression :P

Comment (sonicaj) [reply iocage_lib/create_utils.py]: Updated ;)

**Review by william-gr — APPROVED** [review]

**Review by rick-mesta — APPROVED** [review]
